Add kafka-connect dashboard

francoisferrand opened the pull request:
<img width="1600" alt="kafka-connect-1-general" src="https://user-images.githubusercontent.com/3909027/209164849-77ba8a61-b2a0-4a1e-9370-5df69592a9cb.png">
<img width="1600" alt="kafka-connect-2-workers" src="https://user-images.githubusercontent.com/3909027/209164743-89be7b5c-89de-4333-941d-f9a420219584.png">
<img width="1600" alt="kafka-connect-3-workers" src="https://user-images.githubusercontent.com/3909027/209164754-7e7b6844-9a9b-45bf-97f7-b5a82ac0f4b3.png">
<img width="1600" alt="kafka-connect-4-connectors" src="https://user-images.githubusercontent.com/3909027/209164765-4e343572-76ba-4101-8aa9-88819416c0f7.png">
<img width="1600" alt="kafka-connect-5-rebalances" src="https://user-images.githubusercontent.com/3909027/209164776-cc42bcd7-1d0d-404d-8b17-8704890f581e.png">
<img width="1600" alt="kafka-connect-6-tasks" src="https://user-images.githubusercontent.com/3909027/209164799-12846203-0e70-408f-9336-d1dd5525c4cd.png">
<img width="1600" alt="kafka-connect-7-task-errors" src="https://user-images.githubusercontent.com/3909027/209164891-3a7edd5d-d7b3-4203-bc15-3f7f4f8c1be8.png">
<img width="1600" alt="kafka-connect-8-source" src="https://user-images.githubusercontent.com/3909027/209164912-e49655af-dd55-483f-9430-8736df5a8636.png">
<img width="1600" alt="kafka-connect-9-sink" src="https://user-images.githubusercontent.com/3909027/209164939-3df276e9-6171-42c0-b9fb-9241bb52c1ec.png">
<img width="1600" alt="kafka-connect-10-system" src="https://user-images.githubusercontent.com/3909027/209164949-f4868080-96df-4a64-ae7e-19dfcb17e97d.png">

Issue: ZENKO-4105

francoisferrand committed:
Add kafka-connect dashboard

Issue: ZENKO-4105

bert-e commented:
# Hello francoisferrand,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand committed:
kafka-connect: copy JMX exporter config in repo

Instead of downloading it from JMX exporter repo, store a copy in the repo.
This is needed to further tweak/customize the mapping, and add the
metrics which are not exported at the moment.

Issue: ZENKO-4105

francoisferrand committed:
kafka-connect: expose rate metrics

JMX `-rate` metrics were not exported to prometheus.

Issue: ZENKO-4105

francoisferrand committed:
kafka-connect: add missing percentage and max metrics

The source-record-active-count-max, offset-commit-success-percentage and
batch-size-max metrics were not mapped.

Issue: ZENKO-4105

francoisferrand committed:
Add kafka-connect dashboard to image

Issue: ZENKO-4105

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4105 contains:


* `2.6.1`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.6.2`


Please check the `Fix Version/s` of ZENKO-4105, or the target
branch of this pull request.

welansari commented on monitoring/kafka-connect/dashboard.py:
```diff
@@ -0,0 +1,820 @@
+    maxValue='${replicas}',
+    reduceCalc='last',
+    targets=[Target(
+        expr='sum(up{job="${job}"})',
```
```suggestion
        expr='sum(up{job="${job}", namespace="${namespace}", instance="${instance}"})',
```

francoisferrand replied:
Not sure we should filter on 'instance' here, as it would be displayed as a warning (yellow/red)...

welansari commented on monitoring/kafka-connect/dashboard.py:
```diff
@@ -0,0 +1,820 @@
+            legendFormat='running'
+        ),
+        Target(
+            expr='sum(' + Metrics.worker_connector_running_task_count() + ')',
```
```suggestion
            expr='sum(' + Metrics.worker_connector_failed_task_count() + ')',
```

welansari commented on monitoring/kafka-connect/dashboard.py:
```diff
@@ -0,0 +1,820 @@
+
+class Metrics:
+    connector_status = metrics.Metric(
+        'kafka_connect_connector_status', 'status', 'instance', job='${job}'
```
```suggestion
        'kafka_connect_connector_status', 'status', 'instance', job='${job}', namespace='${namespace}'
```

same for the other metrics

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

williamlardier reviewed (approved):
LGTM

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4105 contains:


* `2.6.2`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.6.3`


Please check the `Fix Version/s` of ZENKO-4105, or the target
branch of this pull request.

francoisferrand committed:
Bump zenko-operator 1.5.3

Issue: ZENKO-4105

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/2.3`
* `development/2.4`
* `development/2.5`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/2.3`
* `development/2.4`
* `development/2.5`



Please check the status of the associated issue ZENKO-4105.

Goodbye francoisferrand.
